File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        l, r = 0, 0
        for i in range(len(nums)):
            if nums[r] > nums[l]:
                nums[r], nums[l+1] = nums[l+1], nums[r]
                l += 1
            r += 1
            
        return l + 1

File: Data Structures & Algorithms/permutations/submission-0.java
class Solution {
    List<List<Integer>> res;
    public List<List<Integer>> permute(int[] nums) {
        res = new ArrayList<>();
        backtrack(new ArrayList<>(), nums, new boolean[nums.length]);
        return res;
    }
    private void backtrack(List<Integer> perm, int[] nums, boolean[]picked){
        if(perm.size() == nums.length){
            res.add(new ArrayList<>(perm));
            return;
        }
        for(int i = 0; i < nums.length; i++){
            if(!picked[i]){
                perm.add(nums[i]);
                picked[i] = true;
                backtrack(perm, nums, picked);
                perm.remove(perm.size() - 1);
                picked[i] = false;
            }
        }

    }
}
